src/main/java/internet/shop/dao/ShoppingCartDao.java

```diff
@@ -1,4 +1,19 @@
 package internet.shop.dao;
 
+import internet.shop.model.ShoppingCart;
+import java.util.List;
+import java.util.Optional;
+
 public interface ShoppingCartDao {
+    ShoppingCart create(ShoppingCart shoppingCart);
+
+    Optional<ShoppingCart> get(Long id);
+
+    List<ShoppingCart> getAll();
+
+    ShoppingCart update(ShoppingCart shoppingCart);
+
+    boolean delete(Long id);
+
+    ShoppingCart getByUserId(Long userId);
```

Comment:
```suggestion
    Optional<ShoppingCart> getByUserId(Long userId);
```
this method is similar to get(Long id)